common/lib/common-utils/src/fluidError.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IFluidError,
+    ITelemetryProperties,
+    ISensitiveDebugData,
+    isIFluidError,
+    ITelemetryBaseEvent,
+} from "@fluidframework/common-definitions";
+
+export abstract class FluidError extends Error implements IFluidError {
```

Comment:
I'm not sure i like how we couple logging and error generation., this can lead to problems where we loose the original stack/error. i'm not sure, but it might be best to seperate them

Reply:
I think I can fix up the problem with losing the original call stack. However, obscuring the original error message is one of the design constraints I'm trying to implement here.

The premise is that **we can only safely log properties on error objects we created**. So this leads me to definitely couple logging and error generation.

Happy to revisit that premise... I feel it's a bit paranoid but I've had conversations with @vladsud, @curtisman, @AndreiZe and others about starting on the conservative side to protect from privacy leaks, and then carefully dial up what/when/how we collect to ensure we can debug effectively.  See #4908 in particular.

When this switches to real PR(s) not draft, I may start by preserving existing behavior.  But I'm trying to prove that the design supports the more restrictive approach.